pkg/auth/dependency/provider/password/password.go

```diff
@@ -14,7 +14,9 @@ const providerPassword string = "password"
 
 type Provider interface {
 	CreatePrincipal(principal Principal) error
-	GetPrincipal(authData interface{}, principal *Principal) error
+	GetPrincipalByAuthData(authData interface{}, principal *Principal) error
```

Comment:
In both login and signup, the `AuthData` is `map[string]interface{}` not `interface{}. We should go for a more specific one here.

Reply:
I used `interface{}` instead of `map[string]interface{}` because I thought what this function requires is something that can be transformed to json.

Reply:
I think the interface here should not concern about the storing format.

Re the authData is `interface` or `map[string]interface{}`, I think it depends on our actual system and use-case. I think in our specification, it must be a `map[string]interface{}` https://github.com/SkygearIO/features/blob/master/features/48-user-profile-discovery/user-profile.md